compiler/modules/nback/scaffold.bal

```diff
@@ -208,6 +209,16 @@ class Scaffold {
         return newDefn;
     }
 
+    function getDecimalString(string str) returns StringDefn {
+        StringDefn? curDefn = self.mod.stringDefns[str];
```

Comment:
Don't use stringDefns. The decimal consts should be nul-terminated without any length.

Reply:
@jclark, I am a bit unclear about this comment. Did you mean we need to model const decimal string as shown below?

`@.str1 = internal unnamed_addr constant [6 x i8] c"2E+10\00", align 8`

Reply:
It's got nothing to do with `.str`. Don't reuse any code of the related to dealing with Ballerina strings. A decimal constant should produce a C style nul-terminated string, so decimal 42 would produce a string with 3 bytes ASCII '4', ASCII '2' and 0.

Reply:
Changed